Create README.md

EgorKhresin opened the pull request.

EgorKhresin committed:
Create README.md

EgorKhresin committed:
подъсоединил шрифты

EgorKhresin committed:
фавиконки

EgorKhresin committed:
постер и ог

EgorKhresin committed:
html  без ховера

EgorKhresin committed:
стили с начатой работой над ховером

EgorKhresin committed:
Rename fonts.css to font.css

EgorKhresin committed:
Update index.html

EgorKhresin committed:
Добавил description для og

EgorKhresin committed:
desription доделываю

EgorKhresin committed:
сделал попап

EgorKhresin committed:
сделал звёздочку

EgorKhresin committed:
сделал звёздочку

EgorKhresin committed:
сделал ховер и вуаль

EgorKhresin committed:
сделал ховер и вуаль

EgorKhresin committed:
поправил смещение текста

EgorKhresin committed:
поправил смещение текста

Savallera commented on index.html:
```diff
@@ -0,0 +1,37 @@
+      name="description"
+      content="Страница о взгляде на перфекционизм в работе и учёбе."
+    />
+    <link rel="icon" href="./icons/icon.ico" sizes="any" />
```
**Надо исправить** https://yadi.sk/d/dJ93XkGcx8kXSw Изображения нужно располагать в соответствующей директории.
Когда файлов в проекте становится много - хорошо их группировать по назначению. Для файлов изображений, например, создать директорию `images` и расположить все изображения там.

Savallera commented on index.html:
```diff
@@ -0,0 +1,37 @@
+    <link rel="stylesheet" href="./styles/style.css" />
+  </head>
+  <body class="page">
+    <video
```
**Можно лучше** Добавьте класс, чтобы можно было описать сво-ва для данного видео, обращение по селектору делает проект менее гибким, при изменении в будущем может понадобится больше времени для внесения правок.

Savallera commented on index.html:
```diff
@@ -0,0 +1,37 @@
+      content="Страница о взгляде на перфекционизм в работе и учёбе."
+    />
+    <meta property="og:title" content="Надо сделать идеально" />
+    <meta property="og:url" content="/" />
```
**Можно лучше** Добавьте страницу GHPages в проект, чтобы можно было описать ее адрес и для портфолио https://habr.com/ru/articles/732546/

EgorKhresin replied:
Огромное спасибо! Я попробовал, это мегакруто))

Savallera commented on index.html:
```diff
@@ -0,0 +1,37 @@
+    <link rel="stylesheet" href="./fonts/font.css" />
+    <link rel="stylesheet" href="./styles/global.css" />
+    <link rel="stylesheet" href="./styles/style.css" />
+  </head>
```
**Отлично** 
  Корректно задан `viewport`, прописан `title` в `head`.

Savallera commented on styles/style.css:
```diff
@@ -0,0 +1,52 @@
+.page::after {
+  content: '';
+  position: absolute;
```
**Надо исправить**
При использовании абсолютного позиционирования необходимо указывать сразу две координаты.
Например, `top` и `left`. Если задавать координату только по одной оси - может возникнуть нестандартное отображение в некоторых браузерах, например в браузере Safari.
Причем даже в последних его версиях.

Savallera commented on styles/global.css:
```diff
@@ -0,0 +1,29 @@
+  grid-template-rows: 100%;
+  min-width: 800px;
+  height: inherit;
+  font-family: Raleway, Arial, sans-serif;
```
**Отлично**
Используется альтернативный шрифт, это полезно, в случае, если браузер не подгрузит основной.

Savallera commented on styles/style.css:
```diff
@@ -0,0 +1,52 @@
+  background-color: #ffffff4d;
+}
+
+.title {
```
**Можно лучше**
Для добавления небольшой интерактивности рекомендуется ссылкам и кнопкам при наведении менять визуально. В этом поможет свойство [`transition`](https://html5book.ru/css3-transition/) и псевдокласс [`hover`](https://doka.guide/css/hover/)

Savallera commented on styles/style.css:
```diff
@@ -0,0 +1,52 @@
+
+.page__title:hover::before {
+  content: 'На самом деле нет.\AВажнее &mdash; научиться ошибаться.';
+  position: absolute;
```
**Можно лучше** У заголовка не указано позиционирование, псевдоэлемент отталкивается от рута

EgorKhresin replied:
Я думал это часть задания:
«Позиционировать заголовок нельзя, нужно оставить его в потоке в исходном состоянии позиционирования — static. Тогда тултип выберет объектом для абсолютного позиционирования ближайшего спозиционированного родителя — body.»
Или я не так понял помарку😥

Savallera replied:
Да, прошу прощенья, я не дочитала про требование, Этот вариант можно занести в хорошие практики, т.к. при наведении на заголовок появляется тултип, чтобы можно было его к нему через относительное позиционирование приклеить.

Savallera commented on styles/style.css:
```diff
@@ -0,0 +1,52 @@
+  background: none;
+}
+
+.page__title:hover::before {
```
**Можно лучше** Добавьте ограничение по мак.ширине, чтобы при дополнении контентом он не вызывал скролл

EgorKhresin replied:
Наверное тоже как-то не так понял. При добавлении максимальной ширины для page__title:hover::before, появляется скролл при добавлении контента и контент выходит за рамки тултипа (если сейчас добавить, то тултип увеличивается и при слишком большом размере начинает выходить за поля без скрола), а максимальная ширина title::after наследуется от заголовка. Прошу прощения, я не понимаю, что именно надо исправить максимальной шириной

Savallera replied:
https://yadi.sk/d/eYC9jx6DD9p5TA без указания макс ширины и со сво-вом white-space, контролировать перенос можно только вручную используя `\A` у псевдоэлементов для переноса на новую строку. 
Если при поддержке или передаче проекта заказчику контент будет несколько больше или изменен, он вылезет за экран или отображение нарушится. В таком случае можно сделать по другому: удалить сво-во white-space и добавить ограничение по максимальной ширине

EgorKhresin replied:
Угу, понял, это тоже было одним из условий задания (свойство white-space и \A)🤓

Savallera reviewed (changes requested):
Здравствуйте!
У Вас получилась хорошая и крепкая работа. Отметила немного замечаний, что можно доделать.
У Вас для этого будет еще одна итерация.
Вы справитесь, желаю вам успехов.

EgorKhresin committed:
Добавил папку с иконками в изображения

Я решил добавить именно папку, а не набор файлов, чтобы было проще и быстрее ориентироваться, если изображений станет больше

EgorKhresin committed:
удалил старую папку с иконками

EgorKhresin committed:
html с новым путём до иконок

EgorKhresin committed:
обновил глобал и стили

EgorKhresin committed:
обновил html

EgorKhresin committed:
обновил og

Savallera commented on index.html:
```diff
@@ -0,0 +1,37 @@
+      content="Страница о взгляде на перфекционизм в работе и учёбе."
+    />
+    <meta property="og:title" content="Надо сделать идеально" />
+    <meta property="og:url" content="https://egorkhresin.github.io/nado_sdelat_idealno/" />
```
Отлично, что применили на практике.

Savallera reviewed (approved):
## Поздравляю, ваша работа принята!

Спасибо вам за терпение и приложенные усилия. У Вас получился замечательный результат. Желаю дальнейших успехов!

---

### Полезности для Вас

**Position**

    - [Особенности позиционирования](http://prgssr.ru/development/5-osobennostej-pozicionirovaniya-v-css.html)
    - [Sticky](https://medium.com/web-standards/sticky-bc7ff7088693)
    - [Почему перемещать элементы с помощью translate лучше, чем с position:absolute top/left](https://habr.com/ru/post/163645/)

**Работа с контентными изображениями**

    - [Object-fit](https://developer.mozilla.org/ru/docs/Web/CSS/object-fit)
    - [Object-fit, object-position](https://css-live.ru/articles/kratkij-obzor-object-fit-i-object-position.html)
    - [Overflow, принципы работы](http://www.css-tricks.ru/articles/css/the-css-overflow-property)
    - [Базовая спека по overflow](https://developer.mozilla.org/ru/docs/Web/CSS/overflow)

**Отрицательные margin**

    - [Отрицательные отступы — путь к универсальности](https://habr.com/ru/post/23748/)
    - [Пример](https://codepen.io/igosheva_nastya/pen/mNRZoa)

**Как работать с посторонними шрифтами**

    - [@font-face](http://htmlbook.ru/css/font-face)
    - [Google шрифты](https://fonts.google.com/)

**iFrame**

    - [Элемент iframe](https://webref.ru/html/iframe)

**Анимации CSS**

    - [Библиотека готовых CSS-анимаций для БЭМ-платформы](https://github.com/bem-contrib/bem-animations)
    - [CSS versus JavaScript animations](https://developers.google.com/web/fundamentals/design-and-ux/animations/css-vs-javascript?hl=ru)

**Z-index**

    - [То, что вам никто не говорил о z-index](https://habr.com/ru/post/166435/)

**SVG**

    - [svg — HTML — Дока](https://doka.guide/html/svg/)
    - [как правильно вставлять svg](https://htmlacademy.ru/blog/video/shorts/short-4)

**Макеты в открытом доступе**

    - [Бесплатные макеты PSD для верстки](http://tpverstak.ru/psd-for-practice/)
    - [Тысячи файлов Photoshop, шаблонов и макетов бесплатно](https://ru.365psd.com/)
    - [Free PSD. +12,800 Photoshop Templates | PSD Repo](https://psdrepo.com/)

**Для того, чтобы использовать макеты:**

    - [avocode](https://avocode.com/)
    - [figma](https://www.figma.com/)
